src/Reflection/ReflectionClass.php

```diff
@@ -490,6 +495,60 @@ public function hasConstant(string $name) : bool
     }
 
     /**
+     * Get the reflection object of the specified class constant.
+     *
+     * Returns null if not specified.
+     *
+     * @param string $name
+     * @return ReflectionClassConstant|null
+     */
+    public function getReflectionConstant(string $name) : ?ReflectionClassConstant
+    {
+        $constants = $this->getReflectionConstants();
+
+        if (!array_key_exists($name, $constants)) {
+            return null;
+        }
+
+        return $constants[$name];
+    }
+
+    /**
+     * Returns array of constants for this class
+     *
+     * @param array $accumulator
+     * @param Node\Stmt $stmt
+     * @return array
+     */
+    private function processReflectionConstants(array $accumulator, Node\Stmt $stmt) : array
+    {
+        if ($stmt instanceof ClassConst) {
+            $const = ReflectionClassConstant::createFromNode($this->reflector, $stmt, $this);
+            $accumulator[$const->getName()] = $const;
+        }
+
+        return $accumulator;
+    }
+
+    /**
+     * Get an array reflection object of the defined constants in this class.
+     *
+     * @return ReflectionClassConstant[]
+     */
+    public function getReflectionConstants() : array
+    {
+        if (null === $this->cachedReflectionConstants) {
+            $this->cachedReflectionConstants = array_reduce(
```

Comment:
> @Ocramius native reflection not provide access for constants inherited from parents. Current behavior is exactly the same as in the native reflection.

We should provide a "clean" implementation in our own code. For instance, we have [`Roave\BetterReflection\Reflection\ReflectionClass#getImmediateMethods()`](https://github.com/Roave/BetterReflection/blob/8bdd09ce8feadb027a93468e2da5a7f281eba2e5/src/Reflection/ReflectionClass.php#L378) for methods implemented only at the current inheritance level (without considering ancestors), and then we provide a [reflection-compatible API](https://github.com/Roave/BetterReflection/blob/8bdd09ce8feadb027a93468e2da5a7f281eba2e5/src/Reflection/Adapter/ReflectionClass.php#L132-L144) that retrieves all methods, exactly like internal reflection. The same should apply here.

 > And traits doesn't support constants.

Didn't know this, thanks for clarifying!